### PR TITLE
fix(client): import TFunction from i18next for challenges context

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Ref https://github.com/freeCodeCamp/freeCodeCamp/pull/49800

TFunction is local type in react-i18next, so this aims to import it from where it's valid

### CODE REVIEW

**Review by huyenltnguyen — COMMENTED** [review]
Just a thought: Instead of passing `t` as a prop and having to add a type to it, could we use the `useTranslation` hook from `react-i18next` and pull the `t` function out from there? 🤔 

(Obviously this would only apply to functional components)

**Comment by Sboonny** [comment]
Oliver have the same idea, I haven't implemented it, because I want to create sandbox and test what is the difference between passing a prop and importing a hook 🤔.

To be honest, I don't mind either, I just want to know the difference between both of them

File: client/src/templates/Introduction/components/block.tsx [hunk 1]
```diff
@@ -128,18 +129,14 @@ class Block extends Component<BlockProps> {
...
-    const {
-      expand: expandText,
-      collapse: collapseText
-    }: {
-      expand: string;
-      collapse: string;
-    } = t('intro:misc-text');
+    const blockTitle = [
+      t(`intro:${superBlock}.blocks.${blockDashedName}.title`)
+    ];
```
Comment (naomi-lgbt) [inline client/src/templates/Introduction/components/block.tsx:134]: Why is this an array?

Comment (Sboonny) [reply client/src/templates/Introduction/components/block.tsx]: There is a function that require it to be an array, refactoring that function lead to creating a subcomponent, but I didn't want the PR to stay for a week, then asked to make the changes in the other PR, so I committed worse code.

I don't think there is a reason for it be an array anymore, except me not wanting to create another PR with the same changes later, and I will wait for this to be merged instead.

For context https://github.com/freeCodeCamp/freeCodeCamp/pull/49820/commits/32ee772e6c24c665a04e805747e0025ba5f9d2c3

Comment (Sboonny) [reply client/src/templates/Introduction/components/block.tsx]: Or different answer, I was tunnel visioned again 🤦

**Review by huyenltnguyen — APPROVED** [review]
LGTM 🙂

**Review by moT01 — APPROVED** [review]

**Pull request merged**